Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -324,7 +324,7 @@ private struct _BSONKeyedEncodingContainer<K: CodingKey> : KeyedEncodingContaine
         self.container = container
     }
 
-    public mutating func encodeNil(forKey key: Key) throws { self.container[key.stringValue] = NSNull() }
+    public mutating func encodeNil(forKey key: Key) throws { self.container[key.stringValue] = BSONNull() }
```

Comment:
nit: can we rename this `encodeNull`?

Reply:
`encodeNil` is actually a requirement of the `Encoder` protocol, so we have to keep the name.

Reply:
oh d'oh that's right